src/app/dataset-view/additional-components/metadata-component/components/add-polling-source/add-polling-source.component.ts

```diff
@@ -0,0 +1,214 @@
+            );
+        }
+    }
+
+    public onCheckedPreprocessStep(event: Event): void {
+        const input = event.target as HTMLInputElement;
+        if (input.checked) {
+            this.pollingSourceForm.addControl(
+                SetPollingSourceSection.PREPROCESS,
+                this.fb.group({
+                    kind: this.defaultPreprocessKind,
+                }),
+            );
+        } else {
+            this.pollingSourceForm.removeControl(
+                SetPollingSourceSection.PREPROCESS,
+            );
+        }
+    }
+
+    constructor(
+        private fb: FormBuilder,
+        private createDatasetService: AppDatasetCreateService,
+        private yamlEventService: TemplatesYamlEventsService,
+        private activatedRoute: ActivatedRoute,
+        private modalService: NgbModal,
+        private cdr: ChangeDetectorRef,
+        private processFormService: ProcessFormService,
+    ) {
+        super();
+    }
+
+    ngOnInit(): void {
+        this.trackSubscription(
+            this.createDatasetService.onErrorCommitEventChanges.subscribe(
+                (message: string) => {
+                    this.errorMessage = message;
+                    this.cdr.detectChanges();
+                },
+            ),
+        );
+    }
+
+    public changeStep(step: SetPollingSourceSection): void {
+        this.currentStep = step;
+    }
+
+    public onSubmit(): void {
+        this.processFormService.transformSchema(this.pollingSourceForm);
+        this.processFormService.processFetchOrderControl(
+            this.pollingSourceForm,
+        );
+        this.trackSubscription(
+            this.createDatasetService
+                .commitEventToDataset(
+                    this.getDatasetInfoFromUrl().accountName,
+                    this.getDatasetInfoFromUrl().datasetName,
+                    this.yamlEventService.buildYamlSetPollingSourceEvent(
+                        this.pollingSourceForm.value as Omit<
+                            SetPollingSource,
+                            "__typename"
+                        >,
+                    ),
+                )
+                .subscribe(),
+        );
+    }
+
+    public getDatasetInfoFromUrl(): DatasetInfo {
+        const paramMap: ParamMap = this.activatedRoute.snapshot.paramMap;
+        return {
+            accountName: requireValue(
+                paramMap.get(ProjectLinks.URL_PARAM_ACCOUNT_NAME),
+            ),
+            datasetName: requireValue(
+                paramMap.get(ProjectLinks.URL_PARAM_DATASET_NAME),
+            ),
+        };
+    }
+
+    public onEditYaml(): void {
+        const modalRef: NgbModalRef = this.modalService.open(
+            FinalYamlModalComponent,
+            { size: "lg" },
+        );
+        this.processFormService.transformSchema(this.pollingSourceForm);
+        this.processFormService.processFetchOrderControl(
+            this.pollingSourceForm,
+        );
```

Comment:
It's good that we have a service now, but can we have 1 facade method, like `transformSchema`, so that the component does not have to know about every correction? Inside it should call current 2 steps, next steps will be added in the service, without component taking care of it.

Besides, soon we will have a reverse transformation, as pressing Back does not properly work.

Reply:
Added 1 public fasade method to processFormService.